### PR TITLE
Fix API issues with pconn connections

### PR DESCRIPTION
**Pull request by psiinon**

### CODE REVIEW

File: src/org/parosproxy/paros/core/proxy/ProxyThread.java [hunk 1]
```diff
@@ -451,13 +452,19 @@ protected void processHttp(HttpRequestHeader requestHeader, boolean isSecure) th
...
 			    }
 			}
-
-			if (parentServer.isEnableApi() &&
-					API.getInstance().handleApiRequest(requestHeader, httpIn, httpOut, isRecursive(requestHeader))) {
-				// It was an API request
-				return;
+			if (parentServer.isEnableApi()) {
+				msg = API.getInstance().handleApiRequest(requestHeader, httpIn, httpOut, isRecursive(requestHeader));
+				if (msg != null) {
```
Comment (thc202) [inline src/org/parosproxy/paros/core/proxy/ProxyThread.java]: Should have another if (to just return) for when the `msg` is empty? We shouldn't notify if the API request was not allowed or is already handled.

File: src/org/parosproxy/paros/core/proxy/ProxyThread.java [hunk 2]
```diff
@@ -451,13 +452,19 @@ protected void processHttp(HttpRequestHeader requestHeader, boolean isSecure) th
...
+					ZapGetMethod method = (ZapGetMethod) msg.getUserObject();
```
Comment (thc202) [inline src/org/parosproxy/paros/core/proxy/ProxyThread.java]: Does it worth get/check the method? (It should never be present, the API does not set it?)

**Comment by psiinon** [comment]
Hopefully fixed those comments

**Comment by thc202** [comment]
Thanks!

**Pull request merged**